worker/src/RTC/TransportTuple.cpp

```diff
@@ -82,4 +82,63 @@ namespace RTC
 
 		MS_DUMP("</TransportTuple>");
 	}
+
+	// TODO: Move to .hpp once done.
+	void TransportTuple::SetHash()
+	{
+		const struct sockaddr* localSockAddr  = GetLocalAddress();
+		const struct sockaddr* remoteSockAddr = GetRemoteAddress();
+
+		switch (localSockAddr->sa_family)
+		{
+			case AF_INET:
+			{
+				auto* remoteSockAddrIn = reinterpret_cast<const struct sockaddr_in*>(remoteSockAddr);
+
+				const auto address = ntohl(remoteSockAddrIn->sin_addr.s_addr);
+				const uint64_t port = (ntohs(remoteSockAddrIn->sin_port));
+
+				std::cout << "address  " << std::bitset<64>{address} << std::endl;
+				std::cout << "port     " << std::bitset<64>{port} << std::endl;
+
+				this->hash = address;
+				this->hash |= port << 32;
+
+				break;
+			}
+
+			case AF_INET6:
+			{
+				auto* remoteSockAddrIn6 = reinterpret_cast<const struct sockaddr_in6*>(remoteSockAddr);
+				auto* a = reinterpret_cast<const uint32_t*>(std::addressof(remoteSockAddrIn6->sin6_addr));
+
+				const auto address = a[0] ^ a[1] ^ a[2] ^ a[3];
+				const uint64_t port = ntohs(remoteSockAddrIn6->sin6_port);
+
+				std::cout << "address  " << std::bitset<64>{address} << std::endl;
+				std::cout << "port     " << std::bitset<64>{port} << std::endl;
+
+				this->hash = address;
+				this->hash |= port << 32;
+
+				break;
+			}
+		}
+
+		std::cout << "tmp hash " << std::bitset<64>{this->hash} << std::endl;
+
+		// Override most significant bit with protocol information:
+		// - If UDP, start with 0.
+		// - If TCP, start with 1.
+		if (this->protocol == Protocol::UDP)
+		{
+			this->hash |= 1LL << 63;
```

Comment:
Doing << 63 did NOT work for me. I'll explain it later.

Reply:
Didi you try to move "1LL"?

Reply:
Yes

Reply:
I tried 1LLU AFAIR. It didn't affect bits higher than 15.